easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1137,6 +1241,49 @@ def robot_find_easyconfig(name, version):
 
     return res
 
+def robot_find_minimal_toolchain_of_dependency(dependency, parent_toolchain):
+    """
+    Find the minimal toolchain of a dependency
+
+    @dependency: dependency target dict (long and short module names may not exist yet)
+    @parent_toolchain: toolchain from which to derive the toolchain hierarchy to search
+    @return: minimal toolchain for which an easyconfig exists for this dependency (and matches build_options)
+    """
+    if build_option('use_existing_modules') and not build_option('retain_all_deps'):
+        existing_modules = modules_tool().available()
+    modtool = modules_tool()
+
+    newdep = copy.deepcopy(dependency)
+    toolchain_hierarchy = get_toolchain_hierarchy(parent_toolchain)
```

Comment:
shorter variable names may make this easier to read

* `dependency` -> `dep`
* `toolchain` -> `tc`